### ISSUE TITLE
RDS Instance replacement when upgrading to 6.0

### ISSUE THREAD

**Issue by mikhailshilkov** [pulumi/pulumi-aws#2682]
### What happened?

Upgrading from 5.x to 6.0 causes a replacement in `aws.rds.Instance`.

```
     Type                 Name        Plan        Info
     pulumi:pulumi:Stack  aws-ts-dev              
 +-  └─ aws:rds:Instance  postgresdb  replace     [diff: ~name]

Resources:
    +-1 to replace
    2 unchanged

Do you want to perform this update? details
  pulumi:pulumi:Stack: (same)
    [urn=urn:pulumi:dev::aws-ts::pulumi:pulumi:Stack::aws-ts-dev]
    ++aws:rds/instance:Instance: (create-replacement)
        [id=postgresdbd65b294]
        [urn=urn:pulumi:dev::aws-ts::aws:rds/instance:Instance::postgresdb]
        [provider: urn:pulumi:dev::aws-ts::pulumi:providers:aws::default_5_41_0::5199a248-3c92-4b11-ae4e-dbf860fd2c76 => urn:pulumi:dev::aws-ts::pulumi:providers:aws::default_6_0_0::output<string>]
      ~ name: "airflow" => "airflow"
    +-aws:rds/instance:Instance: (replace)
        [id=postgresdbd65b294]
        [urn=urn:pulumi:dev::aws-ts::aws:rds/instance:Instance::postgresdb]
        [provider: urn:pulumi:dev::aws-ts::pulumi:providers:aws::default_5_41_0::5199a248-3c92-4b11-ae4e-dbf860fd2c76 => urn:pulumi:dev::aws-ts::pulumi:providers:aws::default_6_0_0::output<string>]
      ~ name: "airflow" => "airflow"
    --aws:rds/instance:Instance: (delete-replaced)
        [id=postgresdbd65b294]
        [urn=urn:pulumi:dev::aws-ts::aws:rds/instance:Instance::postgresdb]
        [provider=urn:pulumi:dev::aws-ts::pulumi:providers:aws::default_5_41_0::5199a248-3c92-4b11-ae4e-dbf860fd2c76]
```

### Expected Behavior

No change in preview.

### Steps to reproduce

1. Deploy this program with the latest `^5.0.0` of AWS provider:

```
import * as pulumi from "@pulumi/pulumi";
import * as aws from "@pulumi/aws";
import * as random from "@pulumi/random";

const dbName = "airflow"
const dbUser = "airflow"

const airflowPassword = new random.RandomPassword(`airflow-password`, {
    length: 16,
    special: false,
});

const db = new aws.rds.Instance("postgresdb", {
    engine: "postgres",

    instanceClass: "db.t4g.micro",
    allocatedStorage: 30,

    name: dbName,
    username: dbUser,
    password: airflowPassword.result,

    skipFinalSnapshot: true,
});
```

2. Upgrade to 6.0.
3. Run `pulumi preview`.

### Output of `pulumi about`

```
CLI          
Version      3.74.0
Go Version   go1.20.5
Go Compiler  gc

Plugins
NAME    VERSION
aws     6.0.0
nodejs  unknown
random  4.13.2

Host     
OS       darwin
Version  13.4.1
Arch     arm64

This project is written in nodejs: executable='/opt/homebrew/bin/node' version='v20.4.0'

Current Stack: mikhail-pulumi-corp/aws-ts/dev

TYPE                                        URN
pulumi:pulumi:Stack                         urn:pulumi:dev::aws-ts::pulumi:pulumi:Stack::aws-ts-dev
pulumi:providers:random                     urn:pulumi:dev::aws-ts::pulumi:providers:random::default_4_13_2
random:index/randomPassword:RandomPassword  urn:pulumi:dev::aws-ts::random:index/randomPassword:RandomPassword::airflow-password
pulumi:providers:aws                        urn:pulumi:dev::aws-ts::pulumi:providers:aws::default_5_41_0
aws:rds/instance:Instance                   urn:pulumi:dev::aws-ts::aws:rds/instance:Instance::postgresdb


Found no pending operations associated with dev

Backend        
Name           pulumi.com
URL            https://app.pulumi.com/mikhail-pulumi-corp
User           mikhail-pulumi-corp
Organizations  mikhail-pulumi-corp, pulumi

Dependencies:
NAME            VERSION
@pulumi/aws     6.0.0
@pulumi/pulumi  3.74.0
@pulumi/random  4.13.2
@types/node     16.18.38

Pulumi locates its logs in /var/folders/8g/ntvfz6q153q0mx7k8h7_y1j80000gn/T/ by default
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by mikhailshilkov** [comment]
Replacement doesn't show up if I create the instance with 6.0 in the first place, and then do a preview

**Comment by mikhailshilkov** [comment]
The issue is reproduce in `6.0.0-alpha.1` as well, the same diff

```
~ name: "airflow" => "airflow"
```

**Comment by lukehoban** [comment]
A few questions/notes:
1. Have we identified any other resources that experience this issue - or is it so far only `aws.rds.Instance`?
2. Is the `[provider: urn:pulumi:dev::aws-ts::pulumi:providers:aws::default_5_41_0::5199a248-3c92-4b11-ae4e-dbf860fd2c76 => urn:pulumi:dev::aws-ts::pulumi:providers:aws::default_6_0_0::output<string>]` part of the diff unrelated, and is it expected?
3. The `name` property was removed from the upstream provider in https://github.com/hashicorp/terraform-provider-aws/pull/31232, as part of the v5 release.  It had been deprecated for some time before that and renamed to `dn_name` as of https://github.com/hashicorp/terraform-provider-aws/pull/22668.
